backend/internal/upgrade/upgrade.go

```diff
@@ -178,12 +183,20 @@ func UpgradeChannel(releaseChannel config.ReleaseChannel) (string, string, error
 		upgradeExecName += ".exe"
 	}
 
-	if err := os.Symlink(filepath.Join(channelDir, "bin", "robin"), filepath.Join(config.GetRobinPath(), "bin", linkExecName)); err != nil && !os.IsExist(err) {
+	// delete and recreate symlink
```

Comment:
delete + symlink is almost as fast as symlink (one tiny syscall to delete is not slow obviously) - but this way you end up with accurate links at the end of it